Docker tutorial (#12)
ThundeRatz/ROSGazeboGuide: InstalationGuides/DockerWindows.md

---
layout: default
title: Docker
parent: Guias de Instalação
nav_order: 5
---

# **Como instalar o Docker no Windows 10**

![Docker Desktop Logo](../assets/img/Docker/docker-desktop-instalation/docker-desk-banner1.png){: .d-block .mx-auto}

## **Índice**<!-- omit in toc -->

- [**Observações Iniciais**](#observações-iniciais)
- [**Pré-requisitos**](#pré-requisitos)
- [**Instalação do Docker Desktop**](#instalação-do-docker-desktop)
- [**Possível erro**](#possível-erro)
- [**Verificação**](#verificação)
- [**Observações Finais**](#observações-finais)
- [**Para saber mais**](#para-saber-mais)

## **Observações Iniciais**

Neste tutorial, ensinaremos como instalar o Docker Desktop no Windows 10. Mas antes de tudo, vamos a algumas observações:

> Em primeiro lugar, para versões anteriores ao Windows 10, o Docker oferecia a opção de usar a Toolbox, mas recentemente deixaram de oferecer suporte e estão migrando tudo para o Docker Desktop, que está disponível apenas para Windows 10 até o momento. Caso você tenha uma versão anterior do Windows, não é recomendado utilizar o Docker Desktop, mas instalar o Docker dentro do WSL (ver tutorial aqui) ou em alguma distro do Linux (neste caso poderia ser em uma Virtual Machine (VM) ou com Dual Boot). 

Author: Faltou o link msm aushdashdu

## **Pré-requisitos**

> Assim sendo, para usar o Docker Desktop, é necessário estar no: Windows 10 64-bit: Pro, Enterprise, ou Education (Build 17134 ou superior) - para Windows 10 Home, uma das exigências do Docker é o WSL2. Se você ainda não o instalou, **[clique aqui](/InstalationGuides/WSL.md)** e para mais informações veja os **[Requerimentos para WSL 2 backend](https://docs.docker.com/docker-for-windows/install/#system-requirements-for-wsl-2-backend)**.

Reviewer: Eu achei um pouco confuso aquele - separando o Home, aí tentei reformular, mas aí a informação do build acabei tirando porque acho que ficou confusa na frase também e acho que não é tãaaao importante, não sei, o que acham?

```suggestion
Assim sendo, para usar o Docker Desktop, é necessário estar no: Windows 10 64-bit: Pro, Enterprise, Education e Home, sendo para esse último necessário ter o WSL2 instalado. Se você ainda não o instalou, **[clique aqui](/InstalationGuides/WSL.md)** e para mais informações veja os **[Requerimentos para WSL 2 backend](https://docs.docker.com/docker-for-windows/install/#system-requirements-for-wsl-2-backend)**.
```

Ok, mas surge a pergunta: "como eu verifico qual versão do Windows estou?" 

Para saber basta ir na barra de pesquisar e escrever “winver”, ali mostrará a versão do Windows que você se encontra!

![winver](../assets/img/Docker/docker-desktop-instalation/winver.png){: .d-block .mx-auto}

![version-checkup](../assets/img/Docker/docker-desktop-instalation/version-checkup.png){: .d-block .mx-auto}

Além disso, temos que verificar se a “Virtualização” está habilitada, para isso pesquise na barra de pesquisa: “Gerenciador de Tarefas” ou “Task Manager” e abra a aplicação:

![task-manager](../assets/img/Docker/docker-desktop-instalation/task-manager.png){: .d-block .mx-auto}

Ao abrir o Gerenciador de Tarefas, vá até a aba de "Desempenho" e verifique se a "Virtualização" está habilitada, como mostra a figura abaixo. Se não conseguir visualizar a aba de desempenho, clique no “Mais Detalhes” no canto inferior esquerdo.

![virtualization](../assets/img/Docker/docker-desktop-instalation/virtualization.png){: .d-block .mx-auto}

> Se estiver habilitado podemos prosseguir com a instalação do Docker Desktop Installer, caso contrário recomendo seguir os passos no **[tutorial de Atualizando para o WSL2](/InstalationGuides/WSL.md)**. ~~Acho que já deu pra perceber que é bem importante que você atualize para o WSL2~~.

Reviewer: ```suggestion
Se estiver habilitado podemos prosseguir com a instalação do Docker Desktop Installer, caso contrário recomendo seguir os passos na parte de atualização para o WSL2 no **[tutorial de instalação do WSL](/InstalationGuides/WSL.md)**. ~~Acho que já deu pra perceber que é bem importante que você atualize para o WSL2~~.
```

Feito o download, podemos executar o arquivo e prosseguir com a instalação pelo Instalation Wizard!

Ao executar o arquivo, o Instalation Wizard abre:

![instalation-wizard-1](../assets/img/Docker/docker-desktop-instalation/instalation-wizard-1.png){: .d-block .mx-auto}

Pode deixar as duas caixinhas assinaladas e clicar em "Ok". Com isso a instalação do Docker Desktop é iniciada junto com suas dependências. Se tudo ocorrer bem, você se deparará com a seguinte tela:

![instalation-wizard-2](../assets/img/Docker/docker-desktop-instalation/instalation-wizard-2.png){: .d-block .mx-auto}

Com isso, será necessário reiniciar o sistema para completar a instalação. Para isso, é só clicar no botão azul "Close and restart", como mostra a figura acima. Então dê o restart aí e nos vemos daqui a pouco.

**Pronto!** Se tudo estiver certo, você já está com o Docker instalado na sua máquina! (sons de aplauso).

## **Possível erro**

Após a reinicialização, pode ocorrer o seguinte erro caso não tenha instalado completamente o WSL2:

![wsl2-error](../assets/img/Docker/docker-desktop-instalation/wsl2-error.png){: .d-block .mx-auto}

> Se isso acontecer, é porque você ainda não seguiu o **[tutorial de Atualizando para o WSL2](/InstalationGuides/WSL.md)**. Lá tem as instruções de como fazer a atualização do kernel do linux. Feito isso, clique em "Restart" e pronto! De qualquer forma, **[aqui](https://docs.microsoft.com/pt-br/windows/wsl/install-win10#step-4---download-the-linux-kernel-update-package)** está a docmentação oficial com as instruções e mais informações sobre a atualização necessária.

Reviewer: ```suggestion
Se isso acontecer, é porque você ainda não seguiu a parte de atualização para o WSL2 no **[tutorial de instalação do WSL](/InstalationGuides/WSL.md)**. Lá tem as instruções de como fazer a atualização do kernel do linux. Feito isso, clique em "Restart" e pronto! De qualquer forma, **[aqui](https://docs.microsoft.com/pt-br/windows/wsl/install-win10#step-4---download-the-linux-kernel-update-package)** está a docmentação oficial com as instruções e mais informações sobre a atualização necessária.
```

## **Verificação**

Agora vamos verificar se o Docker foi instalado corretamente!

Para isso, abra o prompt de comando:

![open-prompt](../assets/img/Docker/docker-desktop-instalation/open-prompt.png){: .d-block .mx-auto}

Digite a seguinte linha e dê enter:

```bash
docker -v
```
A versão instalada será mostrada, como na figura:

![check-docker-desktop-version](../assets/img/Docker/docker-desktop-instalation/check-docker-desktop-version.png){: .d-block .mx-auto}

Além disso, podemos testar se o Docker está funcionando corretamente com o Docker Hub.
Para testar isso digite no prompt de comando: 

```bash
docker run hello-world
```

![docker-hello](../assets/img/Docker/docker-desktop-instalation/docker-hello.png){: .d-block .mx-auto}

> Se a saída for que nem a de cima, está tudo **correto**! (mais sons de aplauso) 

Reviewer: Não se se preferem com ou sem ponto, mas é que tem lugar com e tem lugar sem ahushaushasuhsu.

```suggestion
Se a saída for que nem a de cima, está tudo **correto**! (mais sons de aplauso).
```
